### PR TITLE
fixes #18172: manifest type can now be undefined as well

### LINKED ISSUE #18172: 🐛 Bug Report: useTemplateParameterSchema return value has incorrect type
### 📜 Description

Return value from `useTemplateParameterSchema` casts manifest as `TemplateParameterSchema` but it's undefined when the template is loading. The type is incorrect and it causes bugs.

### 👍 Expected behavior

The type for `manifest` should reflect the fact that value can be undefined.

### 👎 Actual Behavior with Screenshots

TypeScript thinks that manifest is always present.

### 👟 Reproduction steps

1. manifest returned from `useTemplateParameterSchema` without a conditional
2. it'll fail with manifest is undefined

### 📃 Provide the context for the Bug.

_No response_

### 🖥️ Your Environment

_No response_

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by ashinsabu**
## Hey, I just made a Pull Request!

Fixes 18172
Casting manifest as TemplateParameterSchema could previously cause issues when its loading and type is undefined.
image

The type can now be undefined as well preventing failures.
image

NOTE: 
* There was another way to fix this by not casting the manifest at all. A similar hook is seen in "/plugins/scaffolder/src/components/TemplatePage/TemplatePage.tsx" at line 54 where the schema is not casted at all which could be reported as another issue if necessary or can be fixed within this PR with another commit.
* ESLint fixed the formatting as well automatically.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: .changeset/strong-crabs-flow.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-scaffolder-react': major
```
Comment (freben) [inline .changeset/strong-crabs-flow.md:2]: Let's make this minor. You are right that it's technically breaking, but it's in the types and arguably basically a bug fix.

**Review by freben — COMMENTED** [review]
Note the build error - you need to run `yarn build:api-reports` and commit that with the PR

File: .changeset/strong-crabs-flow.md (hunk 1 above)
Comment (ashinsabu) [reply .changeset/strong-crabs-flow.md]: 👍 I've made the required changes. Also want to make sure regarding the first point in the NOTE to typecast the schema as needed. Review and let me know if changes are required whenever you're not busy :)

**Comment by ashinsabu** [comment]
I'm not exactly sure why the checks are failing. I've run the api-report generation command.

**Review by freben — APPROVED** [review]

**Comment by ashinsabu** [comment]
Yes. I guess that is odd if I pull again and run the api-report tool it changes it back to this 
image
This is the terminal log.
image

**Pull request merged**